### PR TITLE
Make partial views Shadow DOM v1 ready

### PR DESCRIPTION
**Pull request by tomalec**
StarcounterApps/Images/issues/84
StarcounterApps/Images/issues/72
Starcounter/RebelsLounge/issues/72

### CODE REVIEW

File: src/Images/wwwroot/Images/viewmodels/EditableContentPage.html [hunk 1]
```diff
@@ -30,19 +30,19 @@
...
+    <template is="declarative-shadow-dom">
         <style>
             .content-form__preview {
                 position: relative;
             }
         </style>
         <div class="content-edit">
             <div class="content-form__preview">
-                <content select="[slot='EditableContent/Preview']"></content>
+                <slot name="EditableContent/Preview"></slot>
```
Comment (Mackiovello) [inline src/Images/wwwroot/Images/viewmodels/EditableContentPage.html]: Just a tiny thing, here, preview is written in uppercase while the preview above is in lowercase. Would suggest that all of them are lowercase for consistency.

Comment (tomalec) [inline src/Images/wwwroot/Images/viewmodels/EditableContentPage.html]: I think @warpech once suggested to use uppercase for slot names (`EditableContent/Preview`) to keep it consistent with view-model (<= C# convention), while  `.content-form__preview` is a regular CSS selector, does not have to match with slot anyhow, so I think we should keep those lower case (as CSS convention). 

I don't mind making slot names lowercase, as it is rather part of  HTML world than C#.

@warpech  WDYT?

Comment (warpech) [inline src/Images/wwwroot/Images/viewmodels/EditableContentPage.html]: I don't think that slot names should follow C# class names conventions.

I agree that we should change to lowercase then.

**Comment by tomalec** [comment]
I have changed slot names to match `AppName/lowercase-slot-name`, for example `Images/preview`. Do you think it's fine to leave app name in original case?

**Comment by warpech** [comment]
Good catch. It should surely be the app name. 

I have a slight preference for lowercase slot names.

Btw. According to the spec, are the slot names case-sensitive?

**Comment by tomalec** [comment]
> Btw. According to the spec, are the slot names case-sensitive?

It's not explicitly stated https://dom.spec.whatwg.org/#slot-name, but regardless of that I would say it's a good idea to assume it is.

So, for `CompositionEditor` app should we have:
a. `CompositionEditor/slot-name`
b. `compositioneditor/slot-name`
c. `composition-editor/slot-name`
?

**Comment by warpech** [comment]
I vote for b., you?

I would like to enforce the same for URIs in our apps in future.

**Comment by tomalec** [comment]
I would choose either b or c, c is slightly easier to read, but may not fit individual app author needs.

Needless to mention, that it introduces a constraint for uniqueness of app name.
Those app names seems to be unique "App Name" "AppName" "App-Name", but our joining and camel/dash casing makes them look the same.

**Comment by warpech** [comment]
I think the rule here is "use the same convention for URIs, BEMs and slots".

**Comment by tomalec** [comment]
So b, isn't it? "Lowercase everything, remove spaces, do not add any additional characters"

**Comment by tomalec** [comment]
Could you tell me what is wrong? This is how it looked before (how it looks on current develop)

**Comment by tomalec** [comment]
Ah! now I got it, I have added fix to this PR. However this is not related to V1, this is old bug in V0 polyfill, which we already encountered in couple of places.

I have started sketching cheat sheet for Shadow DOM V0 & V1, native & polyfill tricks
https://gist.github.com/tomalec/a69081ee79ca492f001c71e4bbe98fdf

File: src/Images/wwwroot/Images/css/illustrations.css [hunk 2]
```diff
@@ -11,8 +11,20 @@
     overflow: auto;
 }
 
-    #images-thumbs ::content > img,
-    #images-thumbs ::slotted(img) {
+    /* For Shadow DOM V0 */
+    #images-thumbs > ::content > img,
```
Comment (warpech) [inline src/Images/wwwroot/Images/css/illustrations.css]: Is line 15 needed at all? There is no browser with native V0 without native V1 that we need to support. Am I missing something?

Comment (tomalec) [inline src/Images/wwwroot/Images/css/illustrations.css]: But maybe some day there will be working V0 polyfill, so we could use l15 instead of l27-34.

**Pull request merged**